src/main/java/gov/nci/ppe/services/impl/UserServiceImpl.java

```diff
@@ -676,8 +680,23 @@ private String sendEmailToParticipantAfterCRCWithdrawsPatient(String firstName,
 				newPatient.setCRC(crc);
 				patientOptional = insertNewPatientDetailsFromOpen(newPatient);
 				newUsersList.add(patientOptional.get());
+				raiseInsertOrUpdateParticipantAuditEvent("PatientID", newPatient.getPatientId(), AuditEventType.PPE_INSERT_DATA_FROM_OPEN.name());
 			}else {
+				// If the patient exists, check for any changes in the relationship with Providers and CRC
+				Participant patient = (Participant)patientOptional.get();
+				Set<Provider> existingProviders = patient.getProviders();
+				CRC existingCRC = patient.getCRC();
+				if(existingProviders.size() != providerSet.size() || !providerSet.containsAll(existingProviders)) {
+					patient.setProviders(providerSet);
+				}
```

Comment:
Add audit event here for Provider change - identify old provider as well

Reply:
Implemented the changes